lib/src/components/NxList/NxList.tsx

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright (c) 2019-present Sonatype, Inc.
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+import React, { forwardRef, MutableRefObject, useCallback, useEffect, useRef, useState } from 'react';
+import classnames from 'classnames';
+import NxListText from './NxListText';
+import NxListSubtext from './NxListSubtext';
+import NxListActions from './NxListActions';
+import NxListButtonItem from './NxListButtonItem';
+import NxListLinkItem from './NxListLinkItem';
+import NxListItem from './NxListItem';
+import NxListDescriptionTerm from './NxListDescriptionTerm';
+import NxListDescription from './NxListDescription';
+import { NxH3 } from '../SimpleComponents';
+import { splitOutFirst } from '../../util/childUtil';
+import { NxListProps, NxLoadError, NxLoadingSpinner } from '../..';
+import useMutationObserver from '@rooks/use-mutation-observer';
+import { NxListComponent, nxListPropTypes } from './types';
+
+const mutationObserverConfig = { subtree: false, childList: true, attributes: false, characterData: false };
+
+const NxList = forwardRef<HTMLUListElement, NxListProps>((props: NxListProps, ref) => {
+  const {className, children, bulleted, emptyMessage, isLoading = false, error, retryHandler, ...attrs } = props;
+  const classNames = classnames(className, 'nx-list', {'nx-list--bulleted': bulleted});
+  const [title, otherChildren] = splitOutFirst(NxH3, children);
+  const [isEmpty, setIsEmpty] = useState(false);
+  const ulRef = useRef<HTMLUListElement | null>(null);
+  const emptyListRef = useRef<HTMLLIElement>(null);
+
+  const updateIsEmpty = useCallback(function updateIsEmpty() {
```

Comment:
Aside from the h3 stuff, this is essentially the same logic that's in `NxTableBody` right?  Seems like a good candidate for a custom hook to me.

Reply:
> Aside from the h3 stuff, this is essentially the same logic that's in `NxTableBody` right? Seems like a good candidate for a custom hook to me.

Custom hook has been created and has been implemented in `NxList`.
b5ff344e7aea78832e5d47f66c290af69ff16e3b